### PR TITLE
Add track color control object

### PR DESCRIPTION
**Pull request by Holzhaus**

### CODE REVIEW

File: src/mixer/basetrackplayer.cpp [hunk 1]
```diff
@@ -478,6 +515,26 @@ void BaseTrackPlayerImpl::slotSetReplayGain(mixxx::ReplayGain replayGain) {
...
+
+void BaseTrackPlayerImpl::slotTrackColorChanged(double v) {
+    if (!m_pLoadedTrack) {
+        return;
+    }
+
+    mixxx::RgbColor::optional_t color = std::nullopt;
+    if (v != kNoTrackColor) {
+        auto colorCode = static_cast<mixxx::RgbColor::code_t>(v);
+        if (v != mixxx::RgbColor::validateCode(colorCode)) {
```
Comment (daschuer) [inline src/mixer/basetrackplayer.cpp:530]: you can use isValidCode()

**Review by daschuer — CHANGES_REQUESTED** [review]
Looks already good, what is missing? 
Very similar to my draft branch I have here.

**Comment by Holzhaus** [comment]
> Looks already good, what is missing?
> Very similar to my draft branch I have here.

For some reason the new color isn't always applied to the track in the track table. I have to investigate.

File: src/track/track.h [hunk 2]
```diff
@@ -330,6 +330,7 @@ class Track : public QObject {
     void keyUpdated(double key);
     void keysUpdated();
     void ReplayGainUpdated(mixxx::ReplayGain replayGain);
```
Comment (uklotzde) [inline src/track/track.h:332]: Should fix the naming of this signal in this PR?

File: src/mixer/basetrackplayer.cpp [hunk 3]
```diff
@@ -125,8 +146,7 @@ BaseTrackPlayerImpl::~BaseTrackPlayerImpl() {
 
 TrackPointer BaseTrackPlayerImpl::loadFakeTrack(bool bPlay, double filebpm) {
     TrackPointer pTrack(Track::newTemporary());
-    pTrack->setAudioProperties(
-            mixxx::kEngineChannelCount,
+    pTrack->setAudioProperties(mixxx::kEngineChannelCount,
```
[the remark this answers is not in the document]
Comment (Holzhaus) [reply src/mixer/basetrackplayer.cpp:149]: clang-format

[the remark this answers is not in the document]
Comment (uklotzde) [reply src/mixer/basetrackplayer.cpp:149]: Keeps them on separate lines for me. Did this change recently?

[the remark this answers is not in the document]
Comment (Holzhaus) [reply src/mixer/basetrackplayer.cpp:149]: I think the line length hook might have caused this. I can revert the changes for now.

[the remark this answers is not in the document]
Comment (Holzhaus) [reply src/mixer/basetrackplayer.cpp:149]: The line-length changes do this.

 When I commit this file, clang-format first removes the manual wrapping, then the line-length hook adds wrapping like this. Maybe we need to update our clang-format config.

[the remark this answers is not in the document]
Comment (uklotzde) [reply src/mixer/basetrackplayer.cpp:149]: This is unfortunate.

[the remark this answers is not in the document]
Comment (uklotzde) [reply src/mixer/basetrackplayer.cpp:149]: VSCode doesn't know about the pre-commit hook. It solely relies on .clang-format. A valid formatting according to .clang-format that doesn't exceed the maximum line length should not be modified.

[the remark this answers is not in the document]
Comment (Holzhaus) [reply src/mixer/basetrackplayer.cpp:149]: Hmm, it doesn't always remove it, maybe only when the indentation is incorrect.

[the remark this answers is not in the document]
Comment (Holzhaus) [reply src/mixer/basetrackplayer.cpp]: Apparently the change was not caused by clang-format. Maybe I accidently pressed `J` in vim without noticing. No idea.

File: src/track/track.h (hunk 2 above)
Comment (Holzhaus) [reply src/track/track.h]: Done.

**Comment by Holzhaus** [comment]
Ok, I think this PR is ready now.

File: src/mixer/basetrackplayer.cpp (hunk 1 above)
Comment (daschuer) [reply src/mixer/basetrackplayer.cpp:539]: Ping!

Comment (Holzhaus) [reply src/mixer/basetrackplayer.cpp:539]: Sorry, I fixed it locally then it reset some changes due to the autoformatting issue and this must have been reverted to its previous state by accident.

**Review by daschuer — APPROVED** [review]

**Comment by daschuer** [comment]
LGTM, @uklotzde: merge?

**Pull request merged**